sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -283,6 +283,64 @@ export class ManagementClient extends LinkEntity {
       : undefined;
   }
 
+  private async _makeManagementRequest(
+    request: AmqpMessage,
+    retryTimeoutInMs: number,
+    sendRequestOptions: SendRequestOptions
+  ): Promise<AmqpMessage> {
+    const initOperationStartTime = Date.now();
+    if (!this._isMgmtRequestResponseLinkOpen()) {
+      log.mgmt(
+        "[%s] Acquiring lock to get the management req res link.",
+        this._context.namespace.connectionId
+      );
+
+      const actionAfterTimeout = () => {
+        const desc: string = `The request with message_id "${request.message_id}" timed out. Please try again later.`;
+        const e: Error = {
+          name: "OperationTimeoutError",
+          message: desc
+        };
+
+        throw e;
+      };
+
+      const waitTimer = setTimeout(actionAfterTimeout, retryTimeoutInMs);
+
+      try {
+        await defaultLock.acquire(this.managementLock, () => {
+          return this._init();
+        });
+      } catch (err) {
+        throw err;
+      } finally {
+        clearTimeout(waitTimer);
+      }
+    }
+    // time taken by the init operation
+    const timeTakenByInit = Date.now() - initOperationStartTime;
+    // Left over time
+    sendRequestOptions.timeoutInMs = retryTimeoutInMs - timeTakenByInit;
+
+    try {
+      return await this._mgmtReqResLink!.sendRequest(request, sendRequestOptions);
+    } catch (err) {
+      err = translate(err);
+      const address =
+        this._mgmtReqResLink || this._mgmtReqResLink!.sender.address
```

Comment:
A few lines above we call `this._mgmtReqResLink!.sendRequest`, then why are we guarding against `this._mgmtReqResLink` being null/undefined?

Also, why are we using `this._mgmtReqResLink!.sender.address` instead of `this._mgmtReqResLink!.address`?

Reply:
> why are we using this._mgmtReqResLink!.sender.address instead of this._mgmtReqResLink!.address?

`address` is not a property on `this._mgmtRqstResLink`.

Moreover, I just followed event-hubs for this..
![image](https://user-images.githubusercontent.com/10452642/78060237-09ef4b80-7340-11ea-9a15-53b38387ff52.png)

Maybe I should update this to..
```ts
      const address = this._mgmtReqResLink!.sender.address || "address";
```



One question.. why was `this._mgmtReqResLink!.sender.address` picked over `this._mgmtReqResLink!.receiver.address` here?

Reply:
> address is not a property on this._mgmtRqstResLink.

Sorry, I meant, you can use `this.address`

Not sure why we did that in Event Hubs, we should change that too

Reply:
Updated to `this.address` in both event-hubs and service-bus.